textarea/textarea.go

```diff
@@ -287,7 +287,7 @@ type Model struct {
 
 	// If promptFunc is set, it replaces Prompt as a generator for
 	// prompt strings at the beginning of each line.
-	promptFunc func(line int) string
+	promptFunc func(line int, focused bool) string
```

Comment:
Perhaps we can have a `PromptMetadata` type that encapsulates both `lineIndex` and `focused` and can be used for any additional future metadata.

Reply:
Added a `PromptInfo` type to match `LineInfo`. @aymanbagabas if this looks good to you, go ahead and merge it.